janus_messages/src/lib.rs

```diff
@@ -204,11 +108,8 @@ pub struct Interval {
 }
 
 impl Interval {
-    /// Create a new [`Interval`] from the provided start and duration. Returns an error if the end
-    /// of the interval cannot be represented as a [`Time`].
+    /// Create a new [`Interval`] from the provided start and duration.
     pub fn new(start: Time, duration: Duration) -> Result<Self, Error> {
-        start.add(duration)?;
-
```

Comment:
I deleted this check because the `Time::add` method it uses is now up in `janus_server`. I'm not sure that it's good that we don't verify that start + duration doesn't overflow `Time`, but then technically such a message could be encoded on the wire, and it'd be the responsibility of some other representation of time (such as one introduced by #445) to detect that.

Reply:
I decided to put an overflow check that doesn't depend on `Time::add` back in. It's gross to duplicate the code that checks for overflow, but this way we aren't changing the semantics of anything, and this can be cleaned up properly on `main`.